### PR TITLE
Add openSUSE Leap 15.2 and move Uyuni Master to it

### PR DESCRIPTION
**Pull request by juliogonzalez**
## What does this PR change?

Add openSUSE Leap 15.2 and move Uyuni Master to it

Libvirt image: https://build.opensuse.org/package/show/systemsmanagement:sumaform:images:libvirt/opensuse152
OpenStack iamge: https://build.opensuse.org/package/show/systemsmanagement:sumaform:images:openstack/opensuse152

### CODE REVIEW

File: modules/proxy/main.tf [hunk 1]
```diff
@@ -5,7 +5,7 @@ variable "images" {
     "4.0-released"   = "sles15sp1"
     "4.0-nightly"    = "sles15sp1"
     "head"           = "sles15sp2"
-    "uyuni-master"   = "opensuse151"
+    "uyuni-master"   = "opensuse152"
     "uyuni-released" = "opensuse151"
```
Comment (juliogonzalez) [inline modules/proxy/main.tf:9]: `uyuni-released` will be changed with Uyuni 2020.07 is released.

File: backend_modules/libvirt/base/main.tf [hunk 2]
```diff
@@ -10,6 +10,8 @@ locals {
     opensuse150o = "https://download.opensuse.org/distribution/leap/15.0/jeos/openSUSE-Leap-15.0-JeOS.x86_64-15.0.1-OpenStack-Cloud-Current.qcow2"
     opensuse151  = "https://download.opensuse.org/repositories/systemsmanagement:/sumaform:/images:/libvirt/images/opensuse151.x86_64.qcow2"
     opensuse151o = "https://download.opensuse.org/distribution/leap/15.1/jeos/openSUSE-Leap-15.1-JeOS.x86_64-15.1.0-OpenStack-Cloud-Current.qcow2"
+    opensuse152  = "https://download.opensuse.org/repositories/systemsmanagement:/sumaform:/images:/libvirt/images/opensuse152.x86_64.qcow2"
```
Comment (rjmateus) [inline backend_modules/libvirt/base/main.tf:13]: should we have the sumaform specific image on new OS?  We are planning on moving away from it soon and only let the one that doesn't provide official images, like `sles12sp3` for example.

File: main.tf.uyuni.example [hunk 3]
```diff
@@ -22,15 +22,15 @@ module "base" {
 //  }
 
   // Required images
-  images = ["centos7", "opensuse151", "ubuntu1804"]
+  images = ["centos7", "opensuse151", "opensuse152", "ubuntu1804"]
```
Comment (rjmateus) [inline main.tf.uyuni.example:25]: should `opensuse151` be remove in here, and update to version 15.2 any minion we have defined in her?

File: salt/mirror/mirror.sh.conf [hunk 4]
```diff
@@ -2,6 +2,7 @@ SCC_CREDS="{{ grains.get('cc_username') }}:{{ grains.get('cc_password') }}"
 IMAGES="https://github.com/moio/sumaform-images/releases/download/4.3.0/centos7.qcow2
 https://download.opensuse.org/repositories/systemsmanagement:/sumaform:/images:/libvirt/images/opensuse150.x86_64.qcow2
 https://download.opensuse.org/repositories/systemsmanagement:/sumaform:/images:/libvirt/images/opensuse151.x86_64.qcow2
+https://download.opensuse.org/repositories/systemsmanagement:/sumaform:/images:/libvirt/images/opensuse152.x86_64.qcow2
```
Comment (rjmateus) [inline salt/mirror/mirror.sh.conf:5]: adding this line will depend on the comment above, and decide if we want to add support to sumaform specific images

**Comment by rjmateus** [comment]
I have tested on libvirt and LGTM

File: backend_modules/libvirt/base/main.tf (hunk 2 above)
Comment (moio) [reply backend_modules/libvirt/base/main.tf:13]: I agree we should not create a custom image unless we have a particular reason to.

File: main.tf.uyuni.example (hunk 3 above)
Comment (Bischoff) [reply main.tf.uyuni.example:25]: it's still needed for KVM; that change will be done later

File: backend_modules/libvirt/base/main.tf (hunk 2 above)
Comment (juliogonzalez) [reply backend_modules/libvirt/base/main.tf:13]: OK. Will change it.

Comment (juliogonzalez) [reply backend_modules/libvirt/base/main.tf:13]: But then I will call it already `opensuse152`, as anyway I guess you are goin to the same for the others, right?

Comment (moio) [reply backend_modules/libvirt/base/main.tf:13]: No, please call it `opensuse152o` - all "official" images (in the sense of non-custom-to sumaform) are currently named that way.

Comment (juliogonzalez) [reply backend_modules/libvirt/base/main.tf]: Oh, god, this is going to confuse people @moio.

Now you have one but not the other.

Comment (rjmateus) [reply backend_modules/libvirt/base/main.tf]: This is what is already happening with ubuntu 20 image, we only have support for `ubuntu2004o`

Comment (juliogonzalez) [reply backend_modules/libvirt/base/main.tf]: Yeah, Silvio told me at rocket.chat. So now I added `opensuse152o` only, but keep in mind this is going to confuse people, as for now nobody is using those images for ubuntu16, etc, and I am pretty sure everyone expects having images without the `o` suffix.

We really need to do the switch to only official images soon, and make things standard.

File: backend_modules/null/host/variables.tf [hunk 5]
```diff
@@ -86,7 +86,7 @@ variable "connect_to_additional_network" {
 }
 
 variable "image" {
-  description = "One of: opensuse150, opensuse151, sles11sp4, sles12, sles12sp1, sles12sp2, sles12sp3, sles12sp4, sles15, centos7"
+  description = "One of: opensuse150, opensuse151, opensuse152, sles11sp4, sles12, sles12sp1, sles12sp2, sles12sp3, sles12sp4, sles15, centos7"
```
Comment (rjmateus) [inline backend_modules/null/host/variables.tf:89]: should be `opensuse152o`

File: modules/base/variables.tf [hunk 6]
```diff
@@ -60,6 +60,6 @@ variable "provider_settings" {
 
 variable "images" {
   description = "list of images to be uploaded to the libvirt host, leave default for all"
-  default     = ["centos7", "opensuse150", "opensuse151", "sles15", "sles15sp1", "sles15sp2", "sles11sp4", "sles12", "sles12sp1", "sles12sp2", "sles12sp3", "sles12sp4", "ubuntu1804"]
+  default     = ["centos7", "opensuse150", "opensuse151", "opensuse152", "sles15", "sles15sp1", "sles15sp2", "sles11sp4", "sles12", "sles12sp1", "sles12sp2", "sles12sp3", "sles12sp4", "ubuntu1804"]
```
Comment (rjmateus) [inline modules/base/variables.tf:63]: should be `opensuse152o`

File: salt/mirror/minima.yaml [hunk 7]
```diff
@@ -7,66 +7,7 @@ scc:
   password: {{ grains.get("cc_password") }}
   repo_names:
     # required by testsuite
-    - SLES11-SP1-Pool
```
Comment (rjmateus) [inline salt/mirror/minima.yaml:10]: Do you really want to delete all these repos?

File: salt/mirror/mirror.sh.conf [hunk 8]
```diff
@@ -2,6 +2,7 @@ SCC_CREDS="{{ grains.get('cc_username') }}:{{ grains.get('cc_password') }}"
 IMAGES="https://github.com/moio/sumaform-images/releases/download/4.3.0/centos7.qcow2
 https://download.opensuse.org/repositories/systemsmanagement:/sumaform:/images:/libvirt/images/opensuse150.x86_64.qcow2
 https://download.opensuse.org/repositories/systemsmanagement:/sumaform:/images:/libvirt/images/opensuse151.x86_64.qcow2
+https://download.opensuse.org/distribution/leap/15.2/appliances/openSUSE-Leap-15.2-JeOS.x86_64-kvm-and-xen.qcow2
```
Comment (rjmateus) [inline salt/mirror/mirror.sh.conf:5]: on libvirt base we are defining the direct URL to the image and didn't use the mirror. Should this image be download to the mirror?

File: salt/mirror/minima.yaml (hunk 7 above)
Comment (juliogonzalez) [reply salt/mirror/minima.yaml:10]: crap!!!!!!!

That's what I used for testing.

File: salt/mirror/mirror.sh.conf (hunk 8 above)
Comment (juliogonzalez) [reply salt/mirror/mirror.sh.conf:5]: Ok, then for now I will remove it from here.

But we should really support using the official images from the mirror as well :-)

File: modules/base/variables.tf (hunk 6 above)
Comment (juliogonzalez) [reply modules/base/variables.tf:63]: Fixed, but I'd say you need to add all the other images you mentioned, such as centos6o, sle12sp5o, etc, right? :-D

File: backend_modules/null/host/variables.tf (hunk 5 above)
Comment (juliogonzalez) [reply backend_modules/null/host/variables.tf]: Solved.

File: salt/mirror/minima.yaml (hunk 7 above)
Comment (juliogonzalez) [reply salt/mirror/minima.yaml]: Solved.

File: modules/base/variables.tf (hunk 6 above)
Comment (juliogonzalez) [reply modules/base/variables.tf]: Solved.

Comment (rjmateus) [reply modules/base/variables.tf]: Good point, I'm not sure. Maybe a work to be done when we "promote" the official images.

File: salt/mirror/mirror.sh.conf (hunk 8 above)
Comment (rjmateus) [reply salt/mirror/mirror.sh.conf]: makes totally sense. I think we can do it when we "promote" the official images.

**Comment by juliogonzalez** [comment]
Ok, merging, merging susemanager-ci PR... and let's hope nothing will break :-D

**Pull request merged**